test/pass/simple-soversion.cmake

```diff
@@ -12,13 +12,26 @@ install_dir(${TEST_DIR}/libsimple CMAKE_ARGS -DBUILD_SHARED_LIBS=On -DCMAKE_WIND
 if(NOT WIN32 AND NOT APPLE)
     test_expect_file(${PREFIX}/lib/libsimple.so)
     test_expect_file(${PREFIX}/lib/libsimple.so.1)
-    test_expect_file(${PREFIX}/lib/libsimple.so.1.1.2)
+    if (DEFINED ENV{ROCM_LIBPATCH_VERSION})
+        test_expect_file(${PREFIX}/lib/libsimple.so.1.1.$ENV{ROCM_LIBPATCH_VERSION})
```

Comment:
The test shouldn't check for the ENV variable. It should set the `ENV` variable and then verify the so version was set correctly.

Reply:
I can set the env variable in the simple-soversion.cmake and verify the test. However the approach is dependent on the previous comment on how to pass the ROCM_LIBPATCH_VERSION as a env or cmake command line parameter.

Reply:
I have updated to tests to remove the ENV. Added separate tests to verify the operation by setting the ENV variables. Please check the latest commit.